lib/puppet/provider/openldap_global_conf/olc.rb

```diff
@@ -1,127 +0,0 @@
-require 'tempfile'
```

Comment:
I introduced another type `config_entry` for single, unique entries. It serves a similar purpose - to add configuration attributes to `cn=config`, but one instance only handles one attribute-value pair and its ensure. This type makes it possible to add one attribute more than once. (e.g.: `olcServerId` and `olcAuthzRegexp` in a MMR setup)

`global_conf` instead handles hashes and applies the values of this hash in one atomic modify call but can only handle one value per attribute. I did try to rewrite the original provider to support more than one value per attribute, but this lead to a massively complex type and idempotency bugs.

I renamed it so the names reflect this. (`config_` -> `cn=config`, `entry` => single attribute-value, `hash` => multiple attributes-values)